### PR TITLE
Use Clap

### PR DESCRIPTION
**Pull request by schultyy**
This pull request replaces docopt with Clap to provide a more convenient and user friendly way to specify command line arguments.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -282,24 +251,28 @@ fn get_cargo_token() -> Option<String> {
...
     // adhere to the user's wish,
     // otherwise we decide based on whether we are running in CI.
-    let write_mode = match args.flag_write {
-        None => ci_env_set(),
-        Some(ref flag) => string_to_bool(flag)
-    };
+
+    let mut write_mode = false;
+
+    if args.is_present("write") {
```
[the remark this answers is not in the document]
Comment (badboy) [reply src/main.rs]: Does this leave us with a way to explicitely turn off write mode?

File: src/main.rs [hunk 2]
```diff
@@ -326,17 +299,36 @@ fn main() {
...
+        .about("Crate publishing done right")
+        .arg(Arg::with_name("write")
+             .short("w")
+             .long("write")
+             .help("Write changes to files (default: yes if CI is set, otherwise no).")
+             .takes_value(false))
+        .arg(Arg::with_name("release")
+            .short("r")
+            .long("release")
+            .help("Create release on GitHub and publish on crates.io (only in write mode) [default: yes].")
```
Comment (badboy) [inline src/main.rs]: I think with above code `release` mode is only really enabled if this flag is passed, right?
Also given that it's now a flag without any argument labeling it to have a default seems wrong

File: src/main.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (schultyy) [reply src/main.rs]: No. I discovered that while fixing the broken tests. I'm working on that already to bring back this toggle.

**Comment by schultyy** [comment]
@badboy I restored the old flag behavior because while I was working on the failing tests I realized that my changes would break the existing and expected behavior.

**Comment by schultyy** [comment]
Opened a ticket for structopt: https://github.com/semantic-rs/semantic-rs/issues/127

**Comment by badboy** [comment]
:shipit:

**Pull request merged**